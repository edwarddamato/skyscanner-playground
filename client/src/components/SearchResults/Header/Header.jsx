import React from 'react';
import { BpkGridContainer, BpkGridRow, BpkGridColumn } from 'bpk-component-grid';
import BpkLargeLongArrowRight from 'bpk-component-icon/lg/long-arrow-right';
import { basicPluralise } from '/utils';
import './Header.scss';

const getTravellers = ({adults, children, infants}) => {
  const total = parseInt(adults + children + infants);
  return `${total} ${basicPluralise('traveller', total)}`;
};

class Header extends React.Component {
  render () {
    return (
      <header className="sr_header">
        <BpkGridContainer>
          <BpkGridRow>
            <BpkGridColumn width={12} tabletWidth={12} mobileWidth={12}>
              <div className="sr_header-title">
                <div className="sr_header-location">{this.props.searchParams.originPlace.toUpperCase()}</div>
                <div className="sr_header-arrow"><BpkLargeLongArrowRight /></div>
                <div className="sr_header-location">{this.props.searchParams.destinationPlace.toUpperCase()}</div>
              </div>
              <ul className="sr_header-info-list">
                <li className="sr_header-info-item">
                  {getTravellers(this.props.searchParams)}
                </li>
                <li className="sr_header-info-item">
                  {this.props.searchParams.cabinClass}
                </li>
              </ul>
            </BpkGridColumn>
          </BpkGridRow>
        </BpkGridContainer>
      </header>
    );
  }
}
Header.propTypes = {
  searchParams: React.PropTypes.object
};

export default Header;
