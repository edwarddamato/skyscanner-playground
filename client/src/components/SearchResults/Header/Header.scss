@import "~bpk-mixins";

.sr {
  &_header {
    .bpk-grid__container {
      background-color: $bpk-color-blue-700;
    }

    &-title {
      float: left;
      width: 100%;
    }
    &-location {
      float: left;
      color: $bpk-color-white;
      font-size: 3em;
      line-height: 100%;
    }
    &-arrow {
      float: left;
      margin: 13px 7px 0 7px;
      svg {
        fill: $bpk-color-white;
      }
    }

    &-info {
      &-list {
        float: left;
        width: 100%;
        color: $bpk-color-blue-100;
      }
      &-item {
        float: left;
        &:after {
          content: ',';
          margin-right: .25em;
        }
        &:last-child::after {
          display: none;
        }
      }
    }
  }
}