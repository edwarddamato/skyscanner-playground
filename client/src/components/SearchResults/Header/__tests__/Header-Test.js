import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../';

describe('Search Results, Header', () => {
  it('renders correctly', () => {
    const header = renderer.create(
      <Header searchParams={{
        originPlace: 'MLA',
        destinationPlace: 'LHR',
        cabinClass: 'Economy',
        adults: 2,
        children: 2,
        infants: 0
      }} />
    ).toJSON();
    expect(header).toMatchSnapshot();
  });
});

