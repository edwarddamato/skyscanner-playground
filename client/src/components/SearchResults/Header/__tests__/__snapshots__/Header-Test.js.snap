// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results, Header renders correctly 1`] = `
<header
  className="sr_header"
>
  <div
    className="bpk-grid__container bpk-grid__container--padded"
  >
    <div
      className="bpk-grid__row"
    >
      <div
        className="bpk-grid__column bpk-grid__column--12 bpk-grid__column--mobile-12 bpk-grid__column--tablet-12"
      >
        <div
          className="sr_header-title"
        >
          <div
            className="sr_header-location"
          >
            MLA
          </div>
          <div
            className="sr_header-arrow"
          >
            <svg
              height="24"
              style={
                Object {
                  "height": "1.5rem",
                  "width": "1.5rem",
                }
              }
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.4 19.5l5.7-5.3c.4-.4.7-.9.8-1.5.1-.3.1-.5.1-.7 0-.2 0-.4-.1-.6-.1-.6-.4-1.1-.8-1.5l-5.7-5.3c-.8-.8-2.1-.7-2.8.1-.8.8-.7 2.1.1 2.8l2.7 2.5H5c-1.1 0-2 .9-2 2s.9 2 2 2h9.4l-2.7 2.5c-.5.4-.7 1-.7 1.5s.2 1 .5 1.4c.8.8 2.1.8 2.9.1z"
              />
            </svg>
          </div>
          <div
            className="sr_header-location"
          >
            LHR
          </div>
        </div>
        <ul
          className="sr_header-info-list"
        >
          <li
            className="sr_header-info-item"
          >
            4 travellers
          </li>
          <li
            className="sr_header-info-item"
          >
            Economy
          </li>
        </ul>
      </div>
    </div>
  </div>
</header>
`;
